__tests__/server.test.js

```diff
@@ -39,44 +39,64 @@ describe('app', () => {
-                topic: "mitch",
-                author: "butter_bridge",
-                body: "I find this existence challenging",
-                created_at: 1594329060000,
-                votes: 100,
-            };
+})
+describe('GET /api/topics', () => {
+    test('tests the connection to the GET /api/topics endpoint', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({ body: { rows } }) => {
+                const testArray = [];
+                expect(rows.length).toBe(3);
+                rows.forEach(row =>
+                    testArray.push([row.slug, row.description])
+                )
+                expect(testArray[0].includes('The man, the Mitch, the legend')).toBe(true)
+                expect(testArray[0].includes('mitch')).toBe(true)
+                expect(testArray[1].includes('Not dogs')).toBe(true)
+                expect(testArray[1].includes('cats')).toBe(true)
+                expect(testArray[2].includes('what books are made of')).toBe(true)
+                expect(testArray[2].includes('paper')).toBe(true)
+            })
+    })
+})
+describe('GET /api/articles/:article_id', () => {
+    test('tests the connection to the GET /api/articles/:article_id parametric endpoint', () => {
+        const testObject1 = {
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: 1594329060000,
+            votes: 100,
+        };
 
-            return request(app)
-                .get('/api/articles/1')
-                .expect(200)
-                .then(({ body: { rows } }) => {
-                    expect(rows[0].article_id).toEqual(1)
-                    expect(rows[0].title).toEqual(testObject1.title)
-                    expect(rows[0].topic).toEqual(testObject1.topic)
-                    expect(rows[0].author).toEqual(testObject1.author)
-                    expect(rows[0].body).toEqual(testObject1.body)
-                    expect(rows[0].votes).toEqual(testObject1.votes)
-                })
-        })
-        test('tests the error handling for bad parametric paths', () => {
-            return request(app)
-                .get('/api/articles/13')
-                .expect(404)
-                .then(({ body: { message } }) => {
-                    expect(message).toBe('404: invalid path');
-                })
-        })
-        test('tests the error handling for bad parametric requests', () => {
-            return request(app)
-                .get('/api/articles/thirteen')
-                .expect(400)
-                .then(({ body: { message } }) => {
-                    expect(message).toBe('400: bad request');
-                })
-        })
+        return request(app)
+            .get('/api/articles/1')
+            .expect(200)
+            .then(({ body: { rows } }) => {
```

Comment:
should be sending data back on the key of `article` not `rows`

Reply:
Also the returned article should not be nested in an array aka should come back as `body: {articles: {...}}`

Reply:
Updated, thanks.